core/src/main/scala/org/ensime/core/SemanticHighlighting.scala

```diff
@@ -53,9 +53,15 @@ class SemanticHighlighting(val global: RichPresentationCompiler) extends Compile
         } else if (sym.hasFlag(PARAM)) {
           add(ParamSymbol)
         } else {
+
           if (sym.ownerChain.exists(_.isDeprecated)) {
             add(DeprecatedSymbol)
           }
+
+          if (sym.ownerChain.exists(_.annotations.exists(_.atp.toString().endsWith("deprecating")))) {
```

Comment:
this might be better as an or condition on the above if statement, to avoid double additions.